src/mono/mono/mini/mini-runtime.c

```diff
@@ -2938,6 +2938,11 @@ mono_jit_compile_method_jit_only (MonoMethod *method, MonoError *error)
 static gpointer
 get_ftnptr_for_method (MonoMethod *method, gboolean need_unbox, MonoError *error)
 {
+	if (method->is_generic) {
```

Comment:
if we want to also rule out methods from a generic type definition, then you may want to add `|| mono_class_is_gtd (method->klass)`

Reply:
Yes we should throw if the method is not callable meaning a "true" generic method and a generic method that uses the type parameters from the owning class. It seems like `IsGenericMethod()` should return `true` even if the type parameters come from the owning class, but that's a different problem...

I'll make another pass here. Thanks